staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -750,7 +750,7 @@ func (s *Controller) lockedUpdateLoadBalancerHosts(service *v1.Service, hosts []
 }
 
 func wantsLoadBalancer(service *v1.Service) bool {
-	return service.Spec.Type == v1.ServiceTypeLoadBalancer
+	return service.Spec.Type == v1.ServiceTypeLoadBalancer && service.Spec.LoadBalancerClass == nil
```

Comment:
comment here, like "if LoadBalancerClass is set, the user does not want the default cloud-provider LB"